packages/story-editor/src/components/panels/shared/media/dropDownMenu.js

```diff
@@ -0,0 +1,240 @@
+              rest.onClick?.(evt);
+              open(evt);
+            }}
+          >
+            <DefaultListItemLabelDisplayText
+              forwardedAs="span"
+              size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}
+            >
+              {option.label}
+            </DefaultListItemLabelDisplayText>
+          </DefaultListItemInner>
+        )}
+      />
+    );
+  }
+
+  return (
+    <DefaultListItemInner
+      {...rest}
+      ref={ref}
+      isSelected={isSelected}
+      disabled={option.disabled}
+      aria-disabled={option.disabled}
+    >
+      <DefaultListItemLabelDisplayText
+        forwardedAs="span"
+        size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}
+      >
+        {option.label}
+      </DefaultListItemLabelDisplayText>
+    </DefaultListItemInner>
+  );
+});
+
+CustomItemRenderer.propTypes = {
+  option: DROP_DOWN_ITEM,
+  isSelected: PropTypes.bool,
+};
+
+/**
+ * An icon button that displays a dropdown menu on click.
+ *
+ * @param {Object} props Component props.
+ * @param {boolean} props.display Whether the more icon should be displayed.
+ * @param {boolean} props.isMenuOpen If the dropdown menu is open.
+ * @param {Function} props.onMenuOpen Callback for when menu is opened.
+ * @param {Function} props.onMenuClose Callback for when menu is closed without any selections.
+ * @param {Function} props.onMenuSelected Callback for when menu is closed and an option selected.
+ * @param {Function} props.setParentActive Sets the parent element active.
+ * @param {Array} props.options Menu items.
+ * @param {Object} props.children Children.
+ * @return {null|*} Element or null.
+ */
+function DropDownMenu({
+  display,
+  isMenuOpen,
+  onMenuOpen,
+  onMenuClose,
+  onMenuSelected,
+  setParentActive = noop,
+  options,
+  children,
+}) {
+  const MenuButtonRef = useRef();
+
+  const onClose = useCallback(() => {
+    onMenuClose();
+    MenuButtonRef.current?.focus();
+    // Activeness of the parent is lost then due to blurring, we need to set it active "manually"
+    // to focus back on the original button.
+    setParentActive();
+  }, [onMenuClose, setParentActive]);
+
+  const onMenuItemClick = (evt, value) => {
+    onMenuSelected(value);
+  };
+
+  useRovingTabIndex(
+    { ref: MenuButtonRef.current || null },
+    [],
+    BUTTON_NESTING_DEPTH
+  );
+  const focusCanvas = useFocusCanvas();
+  useKeyDownEffect(MenuButtonRef.current || null, 'tab', focusCanvas, [
+    focusCanvas,
+  ]);
+
+  const listId = useMemo(() => `list-${uuidv4()}`, []);
+  const buttonId = useMemo(() => `button-${uuidv4()}`, []);
+
+  // Keep icon and menu displayed if menu is open (even if user's mouse leaves the area).
+  return (
+    <MenuContainer>
+      <MenuButton
+        type={BUTTON_TYPES.TERTIARY}
+        size={BUTTON_SIZES.SMALL}
+        variant={BUTTON_VARIANTS.SQUARE}
+        ref={MenuButtonRef}
+        onClick={onMenuOpen}
+        aria-label={__('More', 'web-stories')}
```

Comment:
I know you mention in the PR desc about making this more reusable in another PR but thinking it would be a good idea to pass the aria label in a a prop here.

Reply:
That's a good thing to keep in mind for a follow-up PR. Easier to do that all together once needed.